### ISSUE TITLE
[Change Proposal] CSRs for Prims

### ISSUE THREAD

**Issue by msfschaffner** [lowRISC/opentitan#7688]
Some comportable IPs (OTP, Flash) have primitives with CSRs nodes. These CSRs are macro-specific and typically (but not only) DFT / test related. Currently these CSRs are defined in a separate Hjson located in a closed repo, maintained by the Silicon Creator.

![image](https://user-images.githubusercontent.com/41358501/128945813-d197fbb3-06f7-4569-8c7c-495c4af76251.png)

This creates problems with our design and DV tooling, since these tools do not understand the concept of "nested CSR" nodes. For example, the DV environment generates RAL models for both `reg_top 1` and `reg_top 2` in the picture above, and these RAL models alias onto each other, which causes problems with both automated and directed tests.
Furthermore, the crossbar configuration (offset / size) for these CSRs is not inferred correctly by the design and DV tooling.

There are several potential approaches to this problem and discussions on how to solve this are currently ongoing.
However, it is clear that a change (either on the DV, tooling or design side) is going to be necessary.
This issue serves as a place holder until we have converged on a solution.

See also [this slide deck](https://docs.google.com/presentation/d/19XZXqM4AF94y6FMtbfVaqQjlps-Rp5-tHQN7pkuW7I4/) for a summary of the solutions considered so far.

**Comment by asb** [comment]
Hi Michael - just to double-check, is this ready for review from the Technical Committee from your perspective?

**Comment by msfschaffner** [comment]
Hey Alex - yes this is ready. We have put together a CFR document here:
https://docs.google.com/document/d/1pnDU3-V2BwEbw2nxFmSClJTIODKlCGhPS4TFC8n7D04/edit#heading=h.l7qwxjui9ien

Do you want me to send this to a specific mailing list also?

**Comment by asb** [comment]
Nope, tagging it with the RFC label is fine. I've put it on the agenda for the next meeting. We did used to have separate RFC: Proposal and RFC: TC review labels (with the latter indicating it's ready for review), but it seems that label was removed at some point.

**Comment by msfschaffner** [comment]
Alright, thanks!

**Comment by tjaychen** [comment]
@msfschaffner should we close this now? I know you've already implemented a scheme for this and it is already in production.

**Comment by msfschaffner** [comment]
Yeah I am going to close it in a bit. I still need to document some of this.

**Comment by msfschaffner** [comment]
This can go in the M2.5 backlog.
